### PR TITLE
Feature/edit admin

### PR DESCRIPTION
**Pull request by anamaiabap**
Criando a parte de edição das badges já cadastradas
Se quiserem testar na prática, podem acessar : https://ana--sandboxbrdev.myvtex.com/admin/badges na aba de edição

### CODE REVIEW

File: react/Provider/provider.tsx [hunk 1]
```diff
@@ -229,6 +257,140 @@ const Provider: FC = props => {
...
+      }
+    )
+
+    const conditionsValues = {
+      simpleStatements: statementList,
+      operator: valuesSearchBadges[index].operator,
+    }
+
+    setName(valuesSearchBadges[index].name)
+    if (valuesSearchBadges[index].type === 'html') {
```
Comment (wagnerlduarte) [inline react/Provider/provider.tsx:329]: Do you have a enum or type that represents `html` and `text`, no? Try to use them in this cases. It's a better practice than literal `string`.

File: react/Provider/provider.tsx [hunk 2]
```diff
@@ -229,6 +257,140 @@ const Provider: FC = props => {
...
+        variables: { id: editId, saveData: valueSave },
+      })
+
+      if (returnEdit) {
+        refetch()
+        alert('Badge editada com sucesso')
+        setModalEdit(false)
+      } else {
+        alert('Erro ao editar badge')
+      }
```
Comment (wagnerlduarte) [inline react/Provider/provider.tsx:374]: Change alert to `Toasty` here too.

File: react/Components/modalEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import ImageEdit from './imageEdit'
+import Context from '../Context/context'
+
+const ModalEdit: FC = () => {
+  const provider = useContext(Context)
+
+  const getContent = useCallback(button => {
+    if (button === 'image') return <ImageEdit />
+
+    if (button === 'text') return <InputArea name={'text'} />
```
Comment (wagnerlduarte) [inline react/Components/modalEdit.tsx:25]: Use `enum` or `type` in these case. Using a literal `string` isn't a better way to maintain the code in the future.

File: node/resolvers/deleteMasterdata.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  _: unknown,
+  idBadge: { id: string },
+  ctx: Context
+) {
+  const retorno = await ctx.clients.badges
+    .delete(idBadge.id)
+    .then(() => true)
+    .catch((e: any) => {
+      ctx.vtex.logger.error(
+        `Erro ao deletar Badge através do MasterdataV2 ${e}`
```
Comment (wagnerlduarte) [inline node/resolvers/deleteMasterdata.ts:11]: Translate to english this message

File: node/resolvers/updateMasterdata.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    throw new Error('É necessário enviar um simpleStatements')
+  }
+
+  if (type.length === 0) {
+    throw new Error('É necessário enviar um type')
+  }
+
+  if (idBadges === undefined) {
+    throw new Error('É necessário enviar um ID')
+  }
```
Comment (wagnerlduarte) [inline node/resolvers/updateMasterdata.ts:30]: Translate to english these messages

**Review by wagnerlduarte — CHANGES_REQUESTED** [review]

File: react/Provider/provider.tsx [hunk 6]
```diff
@@ -229,6 +257,140 @@ const Provider: FC = props => {
...
+    const returnDelete = await deleteMasterdataMutation({
+      variables: { id: deleteId },
+    })
+
+    if (returnDelete) {
+      alert('Badge removida com sucesso')
+      refetch()
+    } else {
+      alert('Erro ao remover badge')
+    }
```
[the remark this answers is not in the document]
Comment (anamaiabap) [reply react/Provider/provider.tsx:303]: Amei a dica!!

**Review by wagnerlduarte — APPROVED** [review]

**Review by wagnerlduarte — APPROVED** [review]

**Pull request merged**